app/src/main/java/de/westnordost/streetcomplete/quests/toilet_availability/AddToiletAvailability.java

```diff
@@ -36,10 +63,89 @@ public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
 		changes.add("toilets", yesno);
 	}
 
+	@Nullable @Override public Boolean isApplicableTo(Element element) { return null; }
+
+	@Override public boolean download(BoundingBox bbox, final MapDataWithGeometryHandler handler)
+	{
+		List<AddHousenumber.ElementWithGeometry> items = downloadElementsWithoutToilets(bbox);
+		if(items == null) return false;
+		// empty result: We are done
+		if(items.isEmpty()) return true;
+
+		Envelope bounds = null;
+		for (AddHousenumber.ElementWithGeometry item : items)
+		{
+			Envelope addBounds = item.geometry.getEnvelopeInternal();
+			if(bounds == null) bounds = addBounds;
+			else               bounds.expandToInclude(addBounds);
+		}
+		BoundingBox adjustedBbox = JTSConst.toBoundingBox(bounds);
+
+		ArrayList<Point> toiletsPositions = downloadFreeFloatingPositionsWithToilets(adjustedBbox);
+		if(toiletsPositions == null) return false;
+
+		for (AddHousenumber.ElementWithGeometry item : items)
+		{
+			// exclude elements with toilets inside them
+			int index = AddHousenumber.indexOfPointIn(item.geometry, toiletsPositions);
+			if(index != -1)
+			{
+				toiletsPositions.remove(index);
+				continue;
+			}
+
+			handler.handle(item.element, item.elementGeometry);
+		}
+
+		return true;
+	}
+
+	private List<AddHousenumber.ElementWithGeometry> downloadElementsWithoutToilets(BoundingBox bbox)
+	{
+		final List<AddHousenumber.ElementWithGeometry> list = new ArrayList<>();
+		String elementsWithoutToiletsOverpassQuery = getElementsWithoutToiletsOverpassQuery(bbox);
+		boolean success = overpassServer.getAndHandleQuota(elementsWithoutToiletsOverpassQuery, (element, geometry) ->
+		{
+			if (geometry != null)
+			{
+				Geometry g = JTSConst.toGeometry(geometry);
+				if (g.isValid())
+				{
+					AddHousenumber.ElementWithGeometry item = new AddHousenumber.ElementWithGeometry();
+					item.element = element;
+					item.geometry = g;
+					item.elementGeometry = geometry;
+					list.add(item);
+				}
+			}
+		});
+		return success ? list : null;
+	}
+
+	private ArrayList<Point> downloadFreeFloatingPositionsWithToilets(BoundingBox bbox)
+	{
+		final ArrayList<Point> coords = new ArrayList<>();
+		String query = getFreeFloatingToiletsOverpassQuery(bbox);
+		boolean success = overpassServer.getAndHandleQuota(query, (element, geometry) ->
+		{
+			if(geometry != null)
+			{
+				coords.add(JTSConst.toPoint(geometry.center));
+			}
+		});
+		return success ? coords : null;
+	}
+
 	@Override public String getCommitMessage() { return "Add toilet availability"; }
 	@Override public int getIcon() { return R.drawable.ic_quest_toilets; }
+	@Override public int getTitle() { return R.string.quest_toiletAvailability_name_title; }
 	@Override public int getTitle(@NonNull Map<String, String> tags)
 	{
-		return R.string.quest_toiletAvailability_name_title;
+		boolean isRestArea = "rest_area".equals(tags.get("highway"));
```

Comment:
please add the bigger =services, too

Reply:
Ah... I wasn't aware of them. I will add them too